code/modules/vehicles/mecha/mecha_movement.dm

```diff
@@ -148,3 +148,27 @@
 		var/mob/mob_obstacle = obstacle
 		if(mob_obstacle.move_resist <= move_force)
 			step(obstacle, dir)
+
+/obj/vehicle/sealed/mecha/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change = TRUE)
+	. = ..()
+	if(chassis_camera)
+		update_camera_location(old_loc)
+
+/obj/vehicle/sealed/mecha/forceMove(atom/destination)
+	. = ..()
+	//Only bother updating the camera if we actually managed to move
+	if(. && chassis_camera)
+		update_camera_location(destination)
+
+/obj/vehicle/sealed/mecha/proc/do_camera_update(oldLoc)
+	if(!QDELETED(chassis_camera) && oldLoc != get_turf(src))
```

Comment:
Why all the qdeleted checks? If this was an issue you ran into, you should make sure to clear the camera on mech deletion/destruction